### PR TITLE
Click and reveal long project and package combinations

### PR DESCRIPTION
**Pull request by danidoni**
## Before
image

## After
image

### CODE REVIEW

**Comment by danidoni** [comment]
[Fix notification content with long project/package name combinations](https://trello.com/c/LqMYQLdh/1758-fix-notification-content-with-long-project-package-name-combinations)

**Review by krauselukas — COMMENTED** [review]
Generally it looks good, the only issue I see that there is no indicator that this is scroll able, I checked on my phone and if I wouldn't know I would think the text is just cut. Maybe showing little arrows or similar?

**Comment by danidoni** [comment]
> @danidoni Generally it looks good, the only issue I see that there is no indicator that this is scroll able, I checked on my phone and if I wouldn't know I would think the text is just cut. Maybe showing little arrows or similar?

Yeah, I thought about the same but I'm not sure now to do it...

**Comment by danidoni** [comment]
> > @danidoni Generally it looks good, the only issue I see that there is no indicator that this is scroll able, I checked on my phone and if I wouldn't know I would think the text is just cut. Maybe showing little arrows or similar?
> 
> Yeah, I thought about the same but I'm not sure now to do it... Do you have any ideas on how to do it?

@krauselukas suggested to add a shadow over the hidden content to draw attention to the fact that the content can be scrollable horizontally. That change is now present in the PR.

**Review by dmarcoux — CHANGES_REQUESTED** [review]
Using shadows to hint at scrollable content is somewhat weird. The shadows are still visible when the content is under the top navigation, see this screenshot:

artifact

A better solution would be to shorten the text with an ellipsis, then hint at clicking on the text to fully display it.

**Comment by danidoni** [comment]
> Using shadows to hint at scrollable content is somewhat weird. The shadows are still visible when the content is under the top navigation, see this screenshot:
> 
> artifact
> 
> A better solution would be to shorten the text with an ellipsis, then hint at clicking on the text to fully display it.

That is unfortunately not a perfect solution though, because, for example, in the "Latest updates" section in the main page, the "click-to-reveal" would conflict with the link.

**Comment by hennevogel** [comment]
> "click-to-reveal"

Have you tried accessing this page with your *actual* mobile?

**Comment by danidoni** [comment]
> > "click-to-reveal"
> 
> Have you tried accessing this page with your _actual_ mobile?

Yes, I did.

**Comment by danidoni** [comment]
I've switched the implementation to the "click-and-reveal". It actually makes sense here and it looks better.

**Comment by danidoni** [comment]
In the end I've decided to use an ellipsis to signal that the project and package name is longer than the viewport with. No content reveal is implemented in this PR, though.

**Comment by hennevogel** [comment]
> No content reveal is implemented in this PR, though.

This makes little sense right? As you don't know what the notification is about anymore...

**Comment by danidoni** [comment]
> > No content reveal is implemented in this PR, though.
> 
> This makes little sense right? As you don't know what the notification is about anymore...

Okay, let me give it another try to the click and reveal approach...

**Comment by danidoni** [comment]
I think I got it this time. Please review again @hennevogel @dmarcoux @krauselukas !

**Comment by danidoni** [comment]
@dmarcoux detected that the code does not react to viewport changes, so when switching from landscape to portrait, this implementation does not work. I'll see what I can do.

**Comment by danidoni** [comment]
> @dmarcoux detected that the code does not react to viewport changes, so when switching from landscape to portrait, this implementation does not work. I'll see what I can do.

Fixed, now.

File: src/api/app/views/layouts/webui/webui.html.haml [hunk 1]
```diff
@@ -31,7 +31,7 @@
     = javascript_tag do
       = yield :head_javascript
       var _paq = _paq || [];
-      $(function() { #{yield :ready_function} });
+      $(function() { #{yield :ready_function} window.onresize = applySmartOverflow; });
```
Comment (krauselukas) [inline src/api/app/views/layouts/webui/webui.html.haml:34]: @danidoni I would only add this to the views where it is required

Comment (danidoni) [reply src/api/app/views/layouts/webui/webui.html.haml]: Fixed

**Review by krauselukas — APPROVED** [review]

File: src/api/app/views/webui/users/notifications/_notifications_list.html.haml [hunk 2]
```diff
@@ -29,7 +29,7 @@
                     = check_box_tag('notification_ids[]', notification.id, false,
                                     id: "notification_ids_#{notification.id}", class: 'custom-control-input')
                     = label_tag("notification_ids_#{notification.id}", '', class: 'custom-control-label')
-                .col
+                .col-10
```
[the remark this answers is not in the document]
Comment (danidoni) [reply src/api/app/views/webui/users/notifications/_notifications_list.html.haml:32]: Before, the full-length text had the option to wrap onto the next line, reducing the width of the container, and allowing the columns to distribute the available space. Now, the text is on a single-line and it tries to fill as much space as needed giving no space for the rest of the columns. That `.col-10` ensures the first column get at least some space.

**Comment by dmarcoux** [comment]
I gave this a try on my mobile and it's quite nice, although the surface to tap is a bit small with the ellipsis being only 3 dots. I wonder if we could make this even better by tapping on the project/package names to reveal the complete project/package names. What do you think? This would be much easier on mobile. It's not always possible, but here, it is since the project/package names aren't in a link.

**Review by dmarcoux — APPROVED** [review]

**Pull request merged**